artofillusion/polymesh/PolyMesh.java

```diff
@@ -1299,10 +1300,11 @@ public void setSmoothingMethod(int smoothingMethod) {
 	 * @return A duplicate of the winged mesh object
 	 */
 
-	public Object3D duplicate() {
-		PolyMesh mesh = new PolyMesh();
-		mesh.copyObject(this);
-		return mesh;
+        @Override
+	public PolyMesh duplicate() {
```

Comment:
The method here is retired to return `Object3d` As it implements an abstract method of `Object 3d` - it's part of the API.

Reply:
java specification allows to change overriden method return type and this change does not breaks compatibility because Polymesh IS-A Object3D.
PS. SceneCamera acts the same and changes duplicate() method return type

Reply:
What does the change gain is, though?

Reply:
This avoids a lot of unnessesary casts from Polymesh to Polymesh. Like this one...
>> PolyMesh mesh = (PolyMesh) ((PolyMesh) objInfo.object).duplicate();
Under this project there are a lot of same casts over duplicate() calls

Reply:
Something looks wrong with that line anyway - there seems to be an extra cast?

Can you point me at an actual example?

Reply:
See ex. PolyMeshEditorWindow line 1266... there are cast from mesh object  to PolyMesh where mesh is already PolyMesh instance.
In all project there 46 calls to duplicate() such way